document/core/exec/runtime.rst

```diff
@@ -562,16 +562,26 @@ Finally, the following definition of *evaluation context* and associated structu
    \production{(evaluation contexts)} & E &::=&
      [\_] ~|~
      \val^\ast~E~\instr^\ast ~|~
-     \LABEL_n\{\instr^\ast\}~E~\END ~|~
-     \FRAME_n\{\frame\}~E~\END \\
+     \LABEL_n\{\instr^\ast\}~E~\END \\
    \end{array}
 
 .. math::
+   \begin{array}{l}
    \begin{array}{lcl@{\qquad}l}
    S; F; E[\instr^\ast] &\stepto& S'; F'; E[{\instr'}^\ast]
-     & (\iff S; F; \instr^\ast \stepto S'; F'; {\instr'}^\ast) \\
+   \end{array}
+   \\ \qquad
+     (\iff S; F; \instr^\ast \stepto S'; F'; {\instr'}^\ast) \\
+   \begin{array}{lcl@{\qquad}l}
+   S; F_0; \FRAME_n\{F\}~\instr^\ast~\END &\stepto& S'; F_0; \FRAME_n\{F'\}~\instr'^\ast~\END
+   \end{array}
+   \\ \qquad
+     (\iff S; F; \instr^\ast \stepto S'; F'; {\instr'}^\ast) \\
+   \begin{array}{lcl@{\qquad}l}
    S; F; E[\TRAP] &\stepto& S; F; \TRAP
-     & (\iff E \neq [\_]) \\
+   \end{array}
+   \\ \qquad
+     (\iff E \neq [\_] \wedge E \neq \epsilon~[\_]~\epsilon) \\
```

Comment:
Yeah, I think this change to the side condition isn't really meaningful. On one hand, the way these are actually specified as overlapping syntactic productions, both cases are equivalent and thus just confusing. On the other hand, if the grammar was to be reinterpreted as defining disjoint constructions, you'd need a recursive formulation to also cover "\eps (\eps [_] \eps) \eps" and so on.

So I'd prefer to keep the condition as is and rather add a note saying that this also covers cases like the second.

Reply:
I hadn't see the problem you point out (and wanted to avoid two notes in a row). Will tweak.